lib/compiler/src/compiler.rs

```diff
@@ -99,6 +99,24 @@ pub trait Compiler {
     ) -> Option<Result<Vec<u8>, CompileError>> {
         None
     }
+
+    /// Compiles a module into an object file that can be built with libwasmer
+    /// into a stand-alone executable.
+    ///
+    /// It returns the bytes as a `&[u8]` or a [`CompileError`].
+    fn experimental_object_file_compile_module<'data, 'module>(
```

Comment:
I believe we don't need this. Probably we can replicate what the native engine is doing for generating the object file. Thoughts @MarkMcCaskey ?

Reply:
Yeah, I agree, right now the code paths are the same, I thought I was going to have to do different things is why I split it out